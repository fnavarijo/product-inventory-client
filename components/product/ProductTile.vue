<template>
  <NuxtLink :to="`/catalog/${id}`" class="flex">
    <img class="square-256" src="~/assets/img/product.jpg" alt="">
    <div class="flex flex-col ml-5 justify-center">
      <span class="uppercase font-light">{{ category }}</span>
      <span class="text-lg text-dark-gray-600 hover:text-primary-600 transition ease-in duration-100">{{ name }}</span>
      <p class="text-sm">{{ description }}</p>
      <span class="font-bold text-xl text-dark-gray-500">Q {{ price | currency }}</span>
    </div>
  </NuxtLink>
</template>

<script lang="ts">
import Vue from 'vue';

import { ProductContainerBase } from '@/components/product/ProductContainerBase';

export default Vue.extend({
  mixins: [ProductContainerBase],
  props: {
    description: {
      type: String,
      required: true,
    },
  },
});
</script>
