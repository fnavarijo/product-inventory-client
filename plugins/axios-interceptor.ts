import { Context } from '@nuxt/types';

export default function ({ $axios, store }: Context): void {
  $axios.onError((error) => {
    if (error.response) {
      console.error('Response error:', error.response);
    } else if (error.request) {
      console.error('Request error:', error.request);
      store.commit('notification/setMessage', 'Algo ocurrio mientras se hacia la peticion.');
    } else {
      console.error('Something dark happened');
    }
  });
}
