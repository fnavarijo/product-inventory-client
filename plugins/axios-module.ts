import { Context } from '@nuxt/types';

import { initAxios } from '@/api';

export default ({ $axios }: Context) => {
  initAxios({ axios: $axios });
};
