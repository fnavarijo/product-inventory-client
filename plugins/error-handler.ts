import Vue from 'vue';

Vue.config.errorHandler = function () {
  return true;
};
