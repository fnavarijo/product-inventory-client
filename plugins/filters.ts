import Vue from 'vue';

export const currencyFilter = (value: string) =>
  value
    ? parseFloat(value).toFixed(2)
    : '';

Vue.filter('currency', currencyFilter);
