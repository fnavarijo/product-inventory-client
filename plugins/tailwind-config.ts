import Vue from 'vue';
// @ts-ignore
import resolveConfig from 'tailwindcss/resolveConfig';

import tailwindConfig from '@/tailwind.config';

declare module 'vue/types/vue' {
  interface Vue {
    $twConfig(): any
  }
}

Vue.prototype.$twConfig = () => resolveConfig(tailwindConfig);
