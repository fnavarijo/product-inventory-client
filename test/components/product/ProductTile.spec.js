import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils';

import ProductTile from '@/components/product/ProductTile.vue';

import { currencyFilter } from '@/plugins/filters';

const localVue = createLocalVue();
localVue.filter('currency', currencyFilter);

describe('ProductTile component', () => {
  it('should render component with props', () => {
    const wrapper = mount(ProductTile, {
      propsData: {
        category: 'Shoes',
        name: 'Nike Shoes',
        price: 25.50,
        description: 'Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante.',
        id: 1,
      },
      stubs: {
        NuxtLink: RouterLinkStub,
      },
      localVue,
    });

    expect(wrapper.text()).toContain('Shoes');
    expect(wrapper.text()).toContain('Nike Shoes');
    expect(wrapper.text()).toContain('25.50');
    expect(wrapper.text()).toContain('Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante.');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
