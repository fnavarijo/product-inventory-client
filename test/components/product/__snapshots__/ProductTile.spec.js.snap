// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTile component should render component with props 1`] = `
<a class="flex"><img src="~/assets/img/product.jpg" alt="" class="square-256">
  <div class="flex flex-col ml-5 justify-center"><span class="uppercase font-light">Shoes</span> <span class="text-lg text-dark-gray-600 hover:text-primary-600 transition ease-in duration-100">Nike Shoes</span>
    <p class="text-sm">Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante.</p> <span class="font-bold text-xl text-dark-gray-500">Q 25.50</span>
  </div>
</a>
`;
